bench/micro/Bench/Database/LSMTree/Internal/Merge.hs

```diff
@@ -0,0 +1,391 @@
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (0, 4000)
+        }
+    , benchMerge configWord64
+        { name         = "insert-page-x4"  -- 1 page
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (4056, 4056)
+        }
+    , benchMerge configWord64
+        { name         = "insert-page-plus-byte-x4"  -- 1 page + 1 byte
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (4057, 4057)
+        }
+    , benchMerge configWord64
+        { name         = "insert-huge-vals-x4"  -- 1-5 pages
+        , nentries     = map (`div` 10) $ scaled $ replicate 4 1
+        , finserts     = 1
+        , randomValue  = first serialiseValue . randomByteStringR (10_000, 20_000)
+        }
+      -- common UTxO scenarios
+    , benchMerge configUTxO
+        { name         = "utxo-x4"  -- like tiering merge
+        , nentries     = scaled $ replicate 4 1
+        , finserts     = 1
+        , fdeletes     = 1
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4-uneven"
+        , nentries     = scaled [1, 3, 1.5, 2.5]
+        , finserts     = 1
+        , fdeletes     = 1
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4-lastlevel"
+        , nentries     = scaled $ replicate 4 1
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4+1-min-skewed-lastlevel"  -- live levelling merge
+        , nentries     = scaled [1, 1, 1, 1, 4]
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    , benchMerge configUTxO
+        { name         = "utxo-x4+1-max-skewed-lastlevel"  -- live levelling merge
+        , nentries     = scaled [1, 1, 1, 1, 16]
+        , finserts     = 1
+        , fdeletes     = 1
+        , mergeLevel   = Merge.LastLevel
+        }
+    ]
+  where
+    totalentries = 50_000 :: Word64
+
+    -- For each run, a significant part of all possible keys are in use, e.g.
+    -- half of them for 4 runs of equal size. This leads to many collisions.
+    -- Note that each input run still has the full number of elements due to the
+    -- use of 'sampleWithoutReplacement'.
+    -- Warning: Using this with a run of around half of all entries (or more)
+    -- leads  to issues with generating distinct keys for all its entries!
+    randomWord64WithCollisions :: Rnd SerialisedKey
+    randomWord64WithCollisions =
+          first (serialiseKey . Hash.hash64)
+        . uniformR (0, totalentries `div` 2)
+
+    scaled weights =
+      let total = sum weights
+      in [ round (fromIntegral totalentries * w / total :: Double)
+         | w <- weights
+         ]
```

Comment:
I find it a slightly confusing way of defining how many entries are in each run. Could we hardcode the number instead of scaling them according to `totalEntries`? It would also probably useful if we can change the total number of entries for a single scenario without changing it for all of them. `randomWord64WithCollisions` can then get an `totalEntries` argument instead

Reply:
Hardcoding numbers for each run gets a bit cumbersome and makes it harder to see how many entries there are in total. So I still think something like `scaled` is useful. But probably it's better to have `nentries = 50_000 `distributed` [1, 1, 1, 1, 16]`?

Reply:
Yeah, I like that, thanks!